tasks/install-apt.yml

```
@@ -6,33 +6,70 @@
     name:
       - ca-certificates
       - apt-transport-https
+      - gpg
     state: present
 
+- name: Remove old key (apt)
+  become: true
+  ansible.builtin.file:
+    path: '/etc/apt/keyrings/microsoft.asc'
+    state: absent
+
```

This should be moved above installation or removed completely, since at the end the source with deb822 module still uses this path.

```suggestion
- name: Remove old key (apt)
  become: true
  ansible.builtin.file:
    path: '/etc/apt/keyrings/microsoft.asc'
    state: absent

```